examples/barotropicqgql_betaforced.jl

```diff
@@ -66,13 +66,15 @@ seed!(1234) # reset of the random number generator for reproducibility
 nothing # hide
 
 # Next we construct function `calcF!` that computes a forcing realization every timestep.
-# `ArrayType()` function returns the array type appropriate for the device, i.e., `Array` for
-# `dev = CPU()` and `CuArray` for `dev = GPU()`.
+# First we make sure that if `dev=GPU()`, then `CUDA.rand()` function is called for random
+# numbers uniformy distributed between 0 and 1.
+random_uniform = dev==CPU() ? rand : CUDA.rand
+
 function calcF!(Fh, sol, t, clock, vars, params, grid)
-  ξ = exp.(2π * im * rand(eltype(grid), size(sol))) / sqrt(clock.dt)
-  
-  Fh .= ArrayType(dev)(ξ) .* sqrt.(forcing_spectrum)
-  
+  Fh .= sqrt.(forcing_spectrum) .* exp(2π * im * random_uniform(eltype(grid))) ./ sqrt(clock.dt)
```

Comment:
Looks good :-D

Unfortunately you may need different functions for `Fh::Array` and `Fh::CuArray`, because you may need to use `CUDA.sqrt`?

Reply:
OK, there are some still some issues which I will fix.

But on the other hand, `CUDA.sqrt()` wasn't required previously... why it should be now?

Reply:
Oh, I guess you're right:

```julia
julia> using CUDA

julia> a = CuArray(rand(3))
3-element CuArray{Float64,1}:
 0.3974296213138291
 0.46740283369767677
 0.5039702920081996

julia> b = sqrt.(a)
3-element CuArray{Float64,1}:
 0.6304201942465272
 0.6836686578289785
 0.7099086504672272
```